core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -396,6 +396,13 @@ sealed trait AsyncResponse {
   /** Return None if the result code is OK and KeeperException otherwise. */
   def resultException: Option[KeeperException] =
     if (resultCode == Code.OK) None else Some(KeeperException.create(resultCode, path))
+
+  /**
+   * Throw the exception if there is one
+   */
+  def maybeThrow(): Unit = {
+    resultException.foreach(e => throw e)
```

Comment:
I changed this code to throw the exception directly (instead of using `resultException`) to keep stack traces more readable. I also changed the trait to abstract class for the same reason. It removes 4-5 unnecessary frames from the top of the stack.